### ISSUE TITLE
Skip re-bundling "bundled" code on compile

### ISSUE THREAD

**Issue by RShadowhand** [denoland/deno#12670]
Deno version: 
```
deno 1.15.3 (release, x86_64-unknown-linux-gnu)
v8 9.6.180.9
typescript 4.4.2
```

Running `deno compile index.js` on a minified/uglified/obfuscated code to be unminified and properly formatted, which is an undesired result. Below are the examples.

Input code:
![image](https://user-images.githubusercontent.com/1440674/140639652-dc043922-adb0-473a-8e2b-717779630073.png)

Output binary:
![image](https://user-images.githubusercontent.com/1440674/140639627-42e4cdff-c058-48e4-b34b-87d1b76addcf.png)

**Comment by lucacasonato** [comment]
This seems reasonable. If the module passed to compile has no imports, and is not a TS file, we can skip the bundle step.

**Comment by RShadowhand** [comment]
Is there any way to expedite this? Willing to help in anyway I can!

**Comment by kitsonk** [comment]
The tag `help wanted` indicates that it isn't a priority for the core team and looking for contributions from the wider community.

**Comment by lucacasonato** [comment]
@RShadowhand There is an open PR https://github.com/denoland/deno/pull/12687 that will be landed for 1.17.
